src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,46 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    protected String by;
+    private LocalDate deadline;
+
+    /**
+     *
+     * @param description
+     * @param by
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.deadline = LocalDate.parse(by);
+        this.by = by;
+    }
+
+    /**
+     *
+     * @return
+     */
+    @Override
+    public char getTaskType() { return 'D'; }
```

Comment:
Since this is not a getter, it may be worthwhile to add javadocs for this method!